### PR TITLE
TRUNK-5225 Get List of OrderGroups by Patient or Encounter

### PR DESCRIPTION
**Pull request by krizhan**
https://issues.openmrs.org/browse/TRUNK-5225

Added support to get list of OrderGroups by patient or encounter 

- [x] My pull request only contains ONE single commit

- [x] My IDE is configured to follow the code style of this project.

- [x] I have added tests to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] I ran mvn clean package right before creating this pull request and
added all formatting changes to my commit.

- [x] All new and existing tests passed.

- [ ] My pull request is based on the latest changes of the master branch.
No? Unsure? -> execute command git pull --rebase upstream master

### CODE REVIEW

**Comment by dkayiwa** [comment]
Thanks for working on this @krizhan 😊 
Did you get a chance to look at this? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by krizhan** [comment]
Not yet @dkayiwa . Thanks for the link, I'll check it out.

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java [hunk 1]
```diff
@@ -274,6 +274,36 @@ public OrderGroup getOrderGroupByUuid(String uuid) throws DAOException {
 	public OrderGroup getOrderGroupById(Integer orderGroupId) throws DAOException {
 		return (OrderGroup) sessionFactory.getCurrentSession().get(OrderGroup.class, orderGroupId);
 	}
+
+	/**
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java]: Any reason for having two @see annotations?

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java [hunk 2]
```diff
@@ -274,6 +274,36 @@ public OrderGroup getOrderGroupByUuid(String uuid) throws DAOException {
...
+	 */
+	@Override
+	public List<OrderGroup> getOrderGroupsByPatient(Patient patient) throws DAOException {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(OrderGroup.class);
+
+		if (patient != null) {
+			criteria.add(Restrictions.eq("patient", patient));
+		}
+		
+		return criteria.list();
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java]: Is it intentional to load everything when the patient parameter is null?

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java [hunk 3]
```diff
@@ -274,6 +274,36 @@ public OrderGroup getOrderGroupByUuid(String uuid) throws DAOException {
...
+	 */
+	@Override
+	public List<OrderGroup> getOrderGroupsByEncounter(Encounter encounter) throws DAOException {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(OrderGroup.class);
+
+		if (encounter != null) {
+			criteria.add(Restrictions.eq("encounter", encounter));
+		}
+
+		return criteria.list();
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java]: Is it intentional to load everything when the encounter parameter is null?

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java [hunk 4]
```diff
@@ -274,6 +274,36 @@ public OrderGroup getOrderGroupByUuid(String uuid) throws DAOException {
...
+		
+		return criteria.list();
+	}
+
+	/**
+	 * @see OrderDAO#getOrderGroupsByEncounter(Encounter)
+	 * @see org.openmrs.api.OrderService#getAllOrderGroupsByEncounter(Encounter)
+	 */
+	@Override
+	public List<OrderGroup> getOrderGroupsByEncounter(Encounter encounter) throws DAOException {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java]: Do we really throw any DAOException?

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java [hunk 5]
```diff
@@ -274,6 +274,36 @@ public OrderGroup getOrderGroupByUuid(String uuid) throws DAOException {
 	public OrderGroup getOrderGroupById(Integer orderGroupId) throws DAOException {
 		return (OrderGroup) sessionFactory.getCurrentSession().get(OrderGroup.class, orderGroupId);
 	}
+
+	/**
+	 * @see OrderDAO#getOrderGroupsByPatient(Patient)
+	 * @see org.openmrs.api.OrderService#getAllOrderGroupsByPatient(Patient)
+	 */
+	@Override
+	public List<OrderGroup> getOrderGroupsByPatient(Patient patient) throws DAOException {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java]: Do we really throw any DAOException?

File: api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -1041,6 +1041,18 @@ public OrderGroup getOrderGroupByUuid(String uuid) throws APIException {
 	public OrderGroup getOrderGroup(Integer orderGroupId) throws APIException {
 		return dao.getOrderGroupById(orderGroupId);
 	}
+
+	@Override
+	@Transactional(readOnly = true)
+	public List<OrderGroup> getAllOrderGroupsByPatient(Patient patient) throws APIException {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java]: Do we really throw any APIException?

File: api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -1041,6 +1041,18 @@ public OrderGroup getOrderGroupByUuid(String uuid) throws APIException {
...
+		return dao.getOrderGroupsByPatient(patient);
+	}
+
+	@Override
+	@Transactional(readOnly = true)
+	public List<OrderGroup> getAllOrderGroupsByEncounter(Encounter encounter) throws APIException {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java]: Do we really throw any APIException?

**Comment by krizhan** [comment]
Hey @dkayiwa , I much appreciate the feedback for the previous code changes. I updated it again but left the conversations open for you to resolve as you see fit. Let me know if there are any other issues and thanks again for the feedback!

-Krizhan

**Pull request closed without merging**